Lib/enum.py

```diff
@@ -71,13 +86,45 @@ def _break_on_call_reduce(self, proto):
         setattr(obj, '__reduce_ex__', _break_on_call_reduce)
         setattr(obj, '__module__', '<unknown>')
 
+def _iter_bits_lsb(num):
+    while num:
+        b = num & (~num + 1)
+        yield b
+        num ^= b
+
+def bin(num, max_bits=None):
+    """
+    Like built-in bin(), except negative values are represented in
+    twos-compliment, and the leading bit always indicates sign
+    (0=positive, 1=negative).
+
+    >>> bin(10)
+    '0b0 1010'
+    >>> bin(~10)   # ~10 is -11
+    '0b1 0101'
+    """
+
+    ceiling = 2 ** (num).bit_length()
+    if num >= 0:
+        s = _bltin_bin(num + ceiling).replace('1', '0', 1)
```

Comment:
why not leave the builtin `bin()` to its given name, and name the local version `_bin()`?

Reply:
I like this `bin()` better.  It's not exported in `__all__`, and it's not unusual for modules to define their own versions of common operations without a leading `_`.  At some point in the future I may put it in `__all__`.